### PR TITLE
Add power optimization for i.MX8QM/i.MX8QX

### PR DESCRIPTION
**Pull request by Anson-Huang**
Previous implementation for i.MX8QM/i.MX8QX power management does NOT optimize power number for system suspend, cpu suspend etc., this patch set adds power number optimization for it, turn off system resources whenever we can instead of putting them in standby mode ONLY, and restore them after system resume.

### CODE REVIEW

File: plat/imx/imx8qm/imx8qm_psci.c [hunk 1]
```diff
@@ -105,24 +136,148 @@ void imx_domain_suspend(const psci_power_state_t *target_state)
...
+	if (is_local_state_off(CORE_PWR_STATE(target_state))) {
+		sc_pm_set_cpu_resume(ipc_handle,
+			ap_core_index[cpu_id + PLATFORM_CLUSTER0_CORE_COUNT * cluster_id],
+			false, BL31_BASE);
+		sc_pm_req_cpu_low_power_mode(ipc_handle,
+			ap_core_index[cpu_id + PLATFORM_CLUSTER0_CORE_COUNT * cluster_id],
+			SC_PM_PW_MODE_ON, SC_PM_WAKE_SRC_GIC);
+		plat_gic_cpuif_enable();
+	} else {
+		write_scr_el3(read_scr_el3() & (~0x4));
```
Comment (antonio-nino-diaz-arm) [inline plat/imx/imx8qm/imx8qm_psci.c]: Use definitions rather than magic numbers: ``SCR_FIQ_BIT``

File: plat/imx/imx8qx/imx8qx_psci.c [hunk 2]
```diff
@@ -71,22 +113,104 @@ void imx_domain_suspend(const psci_power_state_t *target_state)
...
+
+	if (is_local_state_off(target_state->pwr_domain_state[MPIDR_AFFLVL1]))
+		sc_pm_req_low_power_mode(ipc_handle, SC_R_A35, SC_PM_PW_MODE_ON);
+
+	if (is_local_state_off(target_state->pwr_domain_state[MPIDR_AFFLVL0])) {
+		sc_pm_req_cpu_low_power_mode(ipc_handle, ap_core_index[cpu_id],
+			SC_PM_PW_MODE_ON, SC_PM_WAKE_SRC_GIC);
+		plat_gic_cpuif_enable();
+	} else {
+		write_scr_el3(read_scr_el3() & (~0x4));
```
Comment (antonio-nino-diaz-arm) [inline plat/imx/imx8qx/imx8qx_psci.c]: ``SCR_FIQ_BIT``

**Comment by antonio-nino-diaz-arm** [comment]
IP Review +1

**Comment by antonio-nino-diaz-arm** [comment]
Also, I think we need to clarify how the maintainership of the plat/imx tree works:
```
plat/imx
├── common : Anson-Huang
├── imx7   : bryanodonoghue and niej
├── imx8m  : JackyBai
├── imx8qm : Anson-Huang
└── imx8qx : Anson-Huang
```
Is this correct?

File: plat/imx/imx8qx/imx8qx_psci.c (hunk 2 above)
Comment (Anson-Huang) [reply plat/imx/imx8qx/imx8qx_psci.c]: OK, will replace it to make it easy for reading.

**Comment by Anson-Huang** [comment]
> Also, I think we need to clarify how the maintainership of the plat/imx tree works:
> plat/imx
> ├── common : Anson-Huang
> ├── imx7   : bryanodonoghue and niej
> ├── imx8m  : JackyBai
> ├── imx8qm : Anson-Huang
> └── imx8qx : Anson-Huang
> 
> Is this correct?

Yes, I think it is correct.

**Comment by antonio-nino-diaz-arm** [comment]
jenkins: test this please

**Pull request merged**